### PR TITLE
Added feature flagging to property group nesting

### PR DESCRIPTION
**Pull request by cortneysmithbentley**

### CODE REVIEW

File: packages/property-grid/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bentley/property-grid-react",
-  "version": "1.2.0",
+  "version": "1.2.1",
```
Comment (aruniverse) [inline packages/property-grid/package.json:3]: did you manually change this? if so revert this, and run `rush change`

**Comment by aruniverse** [comment]
Youve added the option to pass in flags, but dont use this anywhere. you need to update the data provider in PropertyGrid.tsx accordingly

File: packages/property-grid/package.json (hunk 1 above)
Comment (cortneysmithbentley) [reply packages/property-grid/package.json:3]: oh right, thanks!

File: packages/property-grid/src/components/PropertyGrid.tsx [hunk 2]
```diff
@@ -105,7 +105,8 @@ export class PropertyGrid extends React.Component<
         props.rulesetId,
         props.enableFavoriteProperties,
       );
-      this._dataProvider.isNestedPropertyCategoryGroupingEnabled = true;
+      if(PropertyGridManager.flags.enablePropertyGroupNesting)
```
Comment (aruniverse) [inline packages/property-grid/src/components/PropertyGrid.tsx:108]: nit pick, but please add brackets

File: packages/property-grid/src/PropertyGridManager.ts [hunk 3]
```diff
@@ -9,24 +9,36 @@ import { I18N } from "@bentley/imodeljs-i18n";
...
+}
+
 /**
  * Entry point for static initialization required by various components used in the package.
  * @public
  */
 export class PropertyGridManager {
   private static _i18n?: I18N;
-
+  private static _featureFlags: PropertyGridMangerFeatureFlags = {
```
Comment (aruniverse) [inline packages/property-grid/src/PropertyGridManager.ts:24]: maybe add a comment, about this being the _featureFlags obj here being the default values

File: packages/property-grid/src/PropertyGridManager.ts [hunk 4]
```diff
@@ -9,24 +9,36 @@ import { I18N } from "@bentley/imodeljs-i18n";
 import { getClassName, UiError } from "@bentley/ui-abstract";
 import { StateManager } from "@bentley/ui-framework";
 
+/** Supported Feature Flags for  */
```
[the remark this answers is not in the document]
Comment (cortneysmithbentley) [reply packages/property-grid/src/PropertyGridManager.ts:12]: image
this one is your fault, lol. but I'll fix it.

**Review by aruniverse — APPROVED** [review]

**Pull request merged**